### ISSUE TITLE
[dart2js] Options for performance.mark and performance.measure

### ISSUE THREAD

**Issue by shuhei** [dart-lang/sdk#48430]
Currently [html_dart2js.dart](https://raw.githubusercontent.com/dart-lang/sdk/main/sdk/lib/html/dart2js/html_dart2js.dart) contains the following signatures for `Performance` and `WorkerPerformance`.

```dart
  void mark(String markName) native;

  void measure(String measureName, String? startMark, String? endMark) native;
```

It would be great if they also support their overloads with option objects ([mark](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark), [measure](https://developer.mozilla.org/en-US/docs/Web/API/Performance/measure)).

* Dart SDK Version (`dart --version`): `Dart SDK version: 2.17.429489715-edge+google3-v2 (google3) (google3) on "linux_x64"`
* Whether you are using Windows, MacOSX, or Linux (if applicable): Linux for compile and Chrome OS for browser
* Whether you are using Chrome, Safari, Firefox, Edge (if applicable): Chrome

**Comment by srujzs** [comment]
I'm not sure the class `WorkerPerformance` exists anymore - I can't seem to find any reference to it in the MDN docs or w3, and the types don't exist in the global scopes of Chrome and Firefox.

The rest should be easy to do, however. Note that with outdated/incorrect APIs, there's a workaround doc using interop we encourage people to use (especially if the bug is complicated to fix): https://github.com/dart-lang/sdk/blob/main/sdk/lib/html/doc/WORKAROUNDS.md.

**Comment by shuhei** [comment]
@srujzs Thanks for looking into this! Looks like `WorkerPerformance.idl` was merged into `Performance.idl` in Blink ([cl](https://chromium-review.googlesource.com/c/chromium/src/+/1026451)). So, it should be alright to assume that they are identical. What do you think of doing this issue's change to both and creating another issue for deprecating `WorkerPerformance` if necessary?

Aside from that, I'll have a look into adding the options.

**Comment by shuhei** [comment]
Ah, is the WORKAROUNDS.md for workarounds in user code instead of SDK?

Now I understand that the IDL (and interfaces generated from IDL) won't be updated for a while. Please ignore my question about WorkerPerformance above.

**Comment by srujzs** [comment]
> Ah, is the WORKAROUNDS.md for workarounds in user code instead of SDK?

Yup! We encourage users to use interop in their code whenever they come across sticky issues in our web libraries. This is mostly to unblock users quicker and allow us to iterate on a future interop-based version of the web libraries. I have a CL for this, there's just some small details to figure out.

You're right that the IDL won't be updated currently, but we can deprecate stale APIs like `WorkerPerformance` using some of our codegen. I'll file a separate issue to track that.

**Comment by shuhei** [comment]
Thanks a lot for resolving this! Also, the workaround worked well for now 😀

**Comment by srujzs** [comment]
Glad to hear and thanks for filing!
